test/show.jl

```diff
@@ -41,27 +39,181 @@ module TestShow
     refstr = """
     4×3 DataFrames.DataFrame
     
-    │ Col # │ Name │ Eltype  │ Missing │ Values          │
-    ├───────┼──────┼─────────┼─────────┼─────────────────┤
-    │ 1     │ A    │ Int64   │ 0       │ 1  …  4         │
-    │ 2     │ B    │ String  │ 0       │ \"x\\\"\"  …  \"ABC\" │
-    │ 3     │ C    │ Float32 │ 0       │ 1.0  …  4.0     │"""
+    │ Col # │ Name │ Eltype  │ Missing │ Values      │
+    ├───────┼──────┼─────────┼─────────┼─────────────┤
+    │ 1     │ A    │ Int64   │ 0       │ 1  …  4     │
+    │ 2     │ B    │ String  │ 0       │ x\"  …  ABC  │
+    │ 3     │ C    │ Float32 │ 0       │ 1.0  …  4.0 │"""
     for a in [true, false]
         io = IOBuffer()
         showcols(io, df, a, true)
         str = String(take!(io))
         @test str == refstr
     end
 
-    io = IOBuffer()
+    srand(1)
+    df_big = DataFrame(rand(25,5))
+
+    io = IOContext(IOBuffer(), :displaysize=>(10,40))
     show(io, df_big)
+    str = String(take!(io.io))
+    @test str == """
+    25×5 DataFrames.DataFrame. Omitted printing of 2 columns
+    │ Row │ x1       │ x2       │ x3       │
+    ├─────┼──────────┼──────────┼──────────┤
+    │ 1   │ 0.236033 │ 0.644883 │ 0.440897 │
+    ⋮
+    │ 24  │ 0.278582 │ 0.241591 │ 0.990741 │
+    │ 25  │ 0.751313 │ 0.884837 │ 0.550334 │"""
+
+    io = IOContext(IOBuffer(), :displaysize=>(10,40))
     show(io, df_big, true)
+    str = String(take!(io.io))
+    @test str == """
+    25×5 DataFrames.DataFrame
+    │ Row │ x1       │ x2       │ x3       │
+    ├─────┼──────────┼──────────┼──────────┤
+    │ 1   │ 0.236033 │ 0.644883 │ 0.440897 │
+    ⋮
+    │ 24  │ 0.278582 │ 0.241591 │ 0.990741 │
+    │ 25  │ 0.751313 │ 0.884837 │ 0.550334 │
+    
+    │ Row │ x4       │ x5       │
+    ├─────┼──────────┼──────────┤
+    │ 1   │ 0.580782 │ 0.138763 │
+    ⋮
+    │ 24  │ 0.762276 │ 0.755415 │
+    │ 25  │ 0.339081 │ 0.649056 │"""
+
+    io = IOContext(IOBuffer(), :displaysize=>(10,40))
```

Comment:
Maybe set the number of rows to a lower value in order to have smaller test and check what happens when not all rows can be shown in a single page? Can also be done in a later PR if you prefer.

Reply:
I believe this is what I check here. I assume 10 rows and 40 columns. And you can see the difference between `show` and `showall`. 
`show` limits the output to fit page height and `showall` does not do that.
They also differ in how they handle *wide* data (not fitting the screen vertically) and we set `allcols` to `true`: `show` does paging and `showall` prints full table ignoring `:displaysize` (which could useful, when e.g. we want to dump `DataFrame` show result to a file).